src/test/fuzz/CMakeLists.txt

```diff
@@ -4,6 +4,14 @@
 
 add_subdirectory(util)
 
+# Visual Studio 2022 version 17.12 introduced a bug
+# that causes an internal compiler error.
+# See:
+#  - https://github.com/bitcoin/bitcoin/issues/31303
+#  - https://developercommunity.visualstudio.com/t/C1001:-Internal-compiler-error-with-st/10830350
+# Fixed in Visual Studio 2022 version 17.14.
+set(compiler_has_bug "$<AND:$<CXX_COMPILER_ID:MSVC>,$<VERSION_GREATER_EQUAL:$<CXX_COMPILER_VERSION>,19.42>,$<VERSION_LESS:$<CXX_COMPILER_VERSION>,19.44>>")
```

Comment:
Question: Where do these version numbers come from?

Reply:
Answer: There is a table on this page https://learn.microsoft.com/en-us/cpp/overview/compiler-versions?view=msvc-170#version-macros